cmd/syncthing/gui_auth.go

```diff
@@ -128,6 +116,64 @@ func basicAuthAndSessionMiddleware(cookieName string, cfg config.GUIConfiguratio
 	})
 }
 
+func auth(username string, password string, cfg config.GUIConfiguration) bool {
+	if cfg.IsAuthModeLDAP() {
+		return authLDAP(username, password, cfg)
+	} else {
+		return authSimple(username, password, cfg.User, cfg.Password)
+	}
+}
+
+func authSimple(username string, password string, configUser string, configPassword string) bool {
```

Comment:
It's not strictly required for this PR, but the auth functions that have the same signature (or could have) could be types that implement an interface, with something that creates the right one based on the config. It's a bit boilerplate-ish and can be easily added after the fact so skip that for now if you think it's easier.

Not doing any further style nitpicking as Audrius is already all over it. :)